File: sdks/python/src/boxsdk/models/shared/enterprise_base.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class EnterpriseBaseTypeEnum(str, Enum):
    ENTERPRISE = "enterprise"


@dataclass_json
@dataclasses.dataclass
class EnterpriseBase:
    r"""EnterpriseBase
    A mini representation of a enterprise, used when
    nested within another resource.
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: Optional[EnterpriseBaseTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

File: java-sdk/lib/src/main/java/models/shared/EnterpriseBase.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * EnterpriseBase
 * A mini representation of a enterprise, used when
 * nested within another resource.
**/
public class EnterpriseBase {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public EnterpriseBase withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public EnterpriseBaseTypeEnum type;
    public EnterpriseBase withType(EnterpriseBaseTypeEnum type) {
        this.type = type;
        return this;
    }
}